homeassistant/components/lcn/services.py

```diff
@@ -56,18 +56,19 @@ class LcnServiceCall:
     def __init__(self, hass):
         """Initialize service call."""
         self.hass = hass
-        self.connections = hass.data[DATA_LCN][CONF_CONNECTIONS]
 
     def get_device_connection(self, service):
-        """Get device connection object."""
-        addr, connection_id = service.data[CONF_ADDRESS]
-        addr = pypck.lcn_addr.LcnAddr(*addr)
-        if connection_id is None:
-            connection = self.connections[0]
-        else:
-            connection = get_connection(self.connections, connection_id)
-
-        return connection.get_address_conn(addr)
+        """Get address connection object."""
+        address, host_id = service.data[CONF_ADDRESS]
+        unique_device_id = generate_unique_id(address)
+
+        for config_entry in self.hass.config_entries.async_entries(DOMAIN):
+            if config_entry.title == host_id:
+                device_connection = get_device_connection(
+                    self.hass, unique_device_id, config_entry
+                )
+                return device_connection
+        raise ValueError("Invalid host ID.")
```

Comment:
Should an invalid address also (unique id not found) also raise or is returning None OK?